.gitlab/scripts/run_performance_benchmarks.sh

```diff
@@ -124,21 +125,25 @@ for executable in "${pika_targets[@]}"; do
     exit_code=$?
     set -e
     if [[ ${exit_code} -ne 0 ]]; then
+        failures=$((failures + 1))
+
         echo "${executable} failed with exit code ${exit_code}"
-    fi
 
-    # Append command and command line options
-    json_add_value_string "${result_file}" "metric.benchmark.command" "${executable}"
-    json_add_value_string "${result_file}" "metric.benchmark.command_options" "${test_opts}"
+        # Append command and command line options
+        json_add_value_string "${result_file}" "metric.benchmark.command" "${executable}"
+        json_add_value_string "${result_file}" "metric.benchmark.command_options" "${test_opts}"
 
-    # Extract name and timing data from raw result file
-    benchmark_name=$(cat "${raw_result_file}" | keep_json | jq '.outputs[0].name')
-    benchmark_series=$(cat "${raw_result_file}" | keep_json | jq '.outputs[0].series')
-    json_add_value_json "${result_file}" "metric.benchmark.name" "${benchmark_name}"
-    json_add_value_json "${result_file}" "metric.benchmark.series" "${benchmark_series}"
+        # Extract name and timing data from raw result file
+        benchmark_name=$(cat "${raw_result_file}" | keep_json | jq '.outputs[0].name')
+        benchmark_series=$(cat "${raw_result_file}" | keep_json | jq '.outputs[0].series')
+        json_add_value_json "${result_file}" "metric.benchmark.name" "${benchmark_name}"
+        json_add_value_json "${result_file}" "metric.benchmark.series" "${benchmark_series}"
 
-    json_merge "${metadata_file}" "${result_file}" "${result_file}"
-    submit_logstash "${result_file}"
+        json_merge "${metadata_file}" "${result_file}" "${result_file}"
+        submit_logstash "${result_file}"
+    fi
 
     index=$((index + 1))
 done
+
+((failures > 0)) && exit 1
```

Comment:
Very good point. It not being true is the entire point of the expression `((failures > 0))` but since we use `set -e` it's going to stop the job right there. I'll change it to something a bit more explicit, this was unnecessarily cute (and clearly buggy). Thank you!

Reply:
How does the updated version look to you: https://github.com/pika-org/pika/compare/a4abdb2656066d4d13cef9f2994b5b78e51d4a76..923f93b255a322ca6669495a73a5fc2b67e0616e?